runner/ollamarunner/runner.go

```diff
@@ -564,25 +556,43 @@ type CompletionResponse struct {
 	Timings Timings `json:"timings"`
 }
 
-func getSamplers(_ CompletionRequest) []sample.Sampler {
-	// TODO(jessegross): Waiting for sampling code
-
-	/*samplingParams.TopK = req.TopK
-	samplingParams.TopP = req.TopP
-	samplingParams.MinP = req.MinP
-	samplingParams.TypicalP = req.TypicalP
-	samplingParams.Temp = req.Temperature
-	samplingParams.RepeatLastN = req.RepeatLastN
-	samplingParams.PenaltyRepeat = req.RepeatPenalty
-	samplingParams.PenaltyFreq = req.FrequencyPenalty
-	samplingParams.PenaltyPresent = req.PresencePenalty
-	samplingParams.Mirostat = req.Mirostat
-	samplingParams.MirostatTau = req.MirostatTau
-	samplingParams.MirostatEta = req.MirostatEta
-	samplingParams.Seed = uint32(req.Seed)
-	samplingParams.Grammar = req.Grammar*/
-
-	return []sample.Sampler{sample.Greedy()}
+func newSampler(req CompletionRequest) sample.Sampler {
+	transforms := []sample.Transform{}
+
+	if req.Temperature != 0 {
+		// Use greedy if temperature is 0
+		transforms = append(transforms, sample.Temperature(req.Temperature))
+	}
+	if req.TopK > 0 {
+		transforms = append(transforms, sample.TopK(req.TopK))
+	}
+	if req.TopP > 0 {
+		transforms = append(transforms, sample.TopP(req.TopP))
+	}
+	if req.MinP > 0 {
+		transforms = append(transforms, sample.MinP(req.MinP))
+	}
+
+	var sampler sample.Sampler
+	if req.Temperature != 0 {
+		sampler = sample.Greedy(transforms...)
+	} else {
+		sampler = sample.Weighted(nil, transforms...)
+	}
+
+	// TODO(parthsareen): Implement other sampling params
+	// samplingParams.TypicalP = req.TypicalP
+	// samplingParams.RepeatLastN = req.RepeatLastN
+	// samplingParams.PenaltyRepeat = req.RepeatPenalty
+	// samplingParams.PenaltyFreq = req.FrequencyPenalty
+	// samplingParams.PenaltyPresent = req.PresencePenalty
+	// samplingParams.Mirostat = req.Mirostat
+	// samplingParams.MirostatTau = req.MirostatTau
+	// samplingParams.MirostatEta = req.MirostatEta
+	// samplingParams.Seed = uint32(req.Seed)
+	// samplingParams.Grammar = req.Grammar
```

Comment:
We should probably remove this comment for now especially given that we may not support all of these forever